src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+                TestPlatformOptions = options,
+                TestSessionInfo = testSessionInfo
+            };
+
+            // TODO: this will need to be via an adapter instead. Because we will get a "live"
+            // implementation rather than our own implementation that we normally have in design
+            // mode client. It probably will even throw on this cast.
+            var upcastCustomLauncher = (ITestHostLauncher3?)customLauncher;
+
+            TestRequestManager?.RunTests(
+                testRunPayload,
+                upcastCustomLauncher,
+                new RunHandlerToEventsRegistrarAdapter(testRunEventsHandler),
+                new ProtocolConfig { Version = _highestSupportedVersion });
+        }
+        catch (Exception ex)
+        {
+            EqtTrace.Error("InProcessVsTestConsoleWrapper.RunTestsWithCustomTestHost: Exception occurred: " + ex);
+            var testRunCompleteArgs = new TestRunCompleteEventArgs(null, false, true, ex, null, null, TimeSpan.MinValue);
+
+            testRunEventsHandler.HandleLogMessage(TestMessageLevel.Error, ex.ToString());
+            testRunEventsHandler.HandleTestRunComplete(testRunCompleteArgs, null, null, null);
+        }
+
+        _testPlatformEventSource.TranslationLayerExecutionStop();
+    }
+
+    /// <inheritdoc/>
+    public void RunTestsWithCustomTestHost(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        ITestRunEventsHandler testRunEventsHandler,
+        ITestHostLauncher? customTestHostLauncher)
+    {
+        RunTestsWithCustomTestHost(
+            testCases,
+            runSettings,
+            options: null,
+            testRunEventsHandler,
+            customTestHostLauncher);
+    }
+
+    /// <inheritdoc/>
+    public void RunTestsWithCustomTestHost(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        TestPlatformOptions? options,
+        ITestRunEventsHandler testRunEventsHandler,
+        ITestHostLauncher? customTestHostLauncher)
+    {
+        RunTestsWithCustomTestHost(
+            testCases,
+            runSettings,
+            options,
+            testSessionInfo: null,
+            testRunEventsHandler,
+            customTestHostLauncher);
+    }
+
+    /// <inheritdoc/>
+    public void RunTestsWithCustomTestHost(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        TestPlatformOptions? options,
+        TestSessionInfo? testSessionInfo,
+        ITestRunEventsHandler testRunEventsHandler,
+        ITestHostLauncher? customTestHostLauncher)
+    {
+        var testCaseList = testCases.ToList();
+        _testPlatformEventSource.TranslationLayerExecutionStart(
+            1,
+            0,
+            testCaseList.Count,
+            runSettings ?? string.Empty);
+
+        try
+        {
+            TestRequestManager?.ResetOptions();
+
+            var shouldLaunchTesthost = true;
```

Comment:
I don't see any other assignment of this true why is it here?

Reply:
Fixed, thanks.

Reply:
Not fixed